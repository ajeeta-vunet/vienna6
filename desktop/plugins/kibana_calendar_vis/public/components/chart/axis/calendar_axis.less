@import (reference) "~ui/styles/variables";

.month-label {
  font-family: sans-serif;
  text-anchor: middle;
  fill: @gray-light;
  &:hover {
    cursor: pointer;
    fill: @vis-axis-title-color;
  }
}

.weekday-label, .meridiem-label {
  font-family: sans-serif;
  fill: @gray-light;
}

.hour-label {
  font-family: sans-serif;
  text-anchor: middle;
  fill: @gray-light;
}