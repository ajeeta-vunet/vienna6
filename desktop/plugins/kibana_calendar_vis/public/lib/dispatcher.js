/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import d3 from 'd3';

export class Dispatcher {
  constructor() {
    this.container = null;
    this.src = null;
    this.dataTarget = null;
    this.config = null;
    this.API = null;
  }

  newMapping() {
    this.src = null;
    this.dataTarget = null;
    return this;
  }

  addAPI(API) {
    this.API = API;
    return this;
  }

  addConfig(config) {
    this.config = config;
    return this;
  }

  addContainer(container) {
    if (container instanceof HTMLElement) {
      this.container = container;
    } else {
      throw new TypeError(`invalid type ${container.toString()}, should be HTMLElement`);
    }

    return this;
  }

  addSource(src) {
    this.src = src;
    return this;
  }

  addDataTarget(dataTarget) {
    if (!/^\[data-.*\]$/.test(dataTarget)) {
      throw new TypeError(`invalid type ${typeof dataTarget}, should be in a form [data-...]`);
    }
    this.dataTarget = dataTarget;
    return this;
  }

  addEvent(event, callback) {
    if (this.source === null || this.dataTarget === null) {
      throw new TypeError(`source or dataTarget is not defined`);
    }

    const selection = d3.selectAll(this.dataTarget);
    const self = this;
    const nodes = this.src !== null ? [].slice.call(this.container.querySelectorAll(this.src)) : [];
    selection.each(function () {
      const element = d3.select(this);
      if (nodes.length !== 0 && !nodes.includes(this)) {
        return;
      }

      if (typeof callback === 'function') {
        const options = {
          API: self.API,
          config: self.config,
          container: self.container,
          source: self.src,
          dataTarget: self.dataTarget
        };
        return element.on(event, callback.bind(this, options));
      }
    });

    return this;
  }

  removeEvent(event) {
    if (this.source === null || this.dataTarget === null) {
      throw new TypeError(`source or dataTarget is not defined`);
    }

    const selection = d3.selectAll(this.dataTarget);
    selection.each(function () {
      const element = d3.select(this);
      element.on(event, null);
    });

    return this;
  }

}
