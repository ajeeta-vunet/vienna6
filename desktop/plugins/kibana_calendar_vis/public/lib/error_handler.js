/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import MarkdownIt from 'markdown-it';
import d3 from 'd3';

const markdownIt = new MarkdownIt({
  html: true,
  linkify: true
});

export const errorContainer = 'visualize-error';

export class CalendarErrorHandler {
  constructor() {
    this.el = undefined;
    this.id = undefined;
  }

  bindEl(el, id) {
    this.el = el;
    this.id = id;
    return this;
  }

  removeError() {
    d3.select(this.el)
      .attr('id', this.id)
      .selectAll(`.${errorContainer}`)
      .remove();
  }

  error(msg) {
    this.removeError();
    const div = d3.select(this.el)
      .append('div')
      .attr('id', this.id)
      // class name needs `chart` in it for the polling checkSize function
      // to continuously call render on resize
      .attr('class', `${errorContainer} chart error`);

    if (msg === 'No results found') {
      div.append('div')
        .attr('class', `text-center ${errorContainer} visualize-chart`)
        .append('div').attr('class', 'item top')
        .append('div').attr('class', 'item')
        .append('h3').html('No data to show');

      div.append('div').attr('class', 'item bottom');
    } else {
      div.append('h4').html(markdownIt.renderInline(msg));
    }
    return div;
  }
}
