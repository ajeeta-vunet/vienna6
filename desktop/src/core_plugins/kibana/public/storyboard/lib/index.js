export { saveStoryboard } from './save_storyboard';
//export { getAppStateDefaults } from './get_app_state_defaults';
