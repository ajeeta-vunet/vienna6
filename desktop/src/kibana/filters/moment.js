define(function (require) {
	var moment = require('moment');
  var _ = require('lodash');

  require('modules')
    .get('kibana/filters')
    .filter('moment', function () {
    return function (datetime, roundUp) {
      var format = 'MMMM Do YYYY, HH:mm:ss.SSS';
      if (moment.isMoment(datetime)) return datetime.format(format);
      if (_.isDate(datetime)) return moment(datetime).format(format);
      return datetime;
    };
  });
});