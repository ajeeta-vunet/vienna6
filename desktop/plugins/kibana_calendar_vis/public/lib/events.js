/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import $ from 'jquery';
import moment from 'moment';

export function highlight({ config, container, dataTarget }) {
  const label = this.getAttribute(dataTarget.replace(/\=".*"/, '').slice(1, -1));

  function justifyOpacity(opacity) {
    const decimalNumber = parseFloat(opacity, 10);
    const fallbackOpacity = 0.5;
    return (0 <= decimalNumber && decimalNumber <= 1) ? decimalNumber : fallbackOpacity;
  }

  const dimming = config.get('visualization:dimmingOpacity');
  $(container).find(dataTarget)
    .css('opacity', 1)
    .not((els, el) => el.getAttribute(dataTarget.slice(1, -1)) === label)
    .css('opacity', justifyOpacity(dimming));
}

export function unHighlight({ container, dataTarget }) {
  const div = $(container);
  $(dataTarget, div).css('opacity', '');
}

export function expandView({ API }) {
  const { timeFilter } = API;
  const { year, month, day } = this.dataset;
  let start;
  let end;
  if (day === undefined) {
    start = moment(new Date(year, month - 1, 1));
    end = moment(new Date(new Date(year, month, 1).getTime() - 1));
  } else {
    start = moment(new Date(year, month - 1, day, 0));
    end = moment(new Date(new Date(year, month - 1, day, 24).getTime() - 1));
  }

  timeFilter.setTime(start, end, 'absolute');
}
