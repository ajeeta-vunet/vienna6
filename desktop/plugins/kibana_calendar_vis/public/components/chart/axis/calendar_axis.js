/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import React from 'react';
import d3 from 'd3';
import moment from 'moment';
import { AxisConfig } from './axis_config';
import { AXIS_SCALE_TYPE, CalendarAxisScale } from './axis_scale';
import './calendar_axis.less';
import { expandView } from '../../../lib/events';

export class CalendarAxis extends React.Component {
  constructor(props) {
    super(props);
    const { gridConfig, axisConfig, type } = this.props;
    this.axisConfig = new AxisConfig(axisConfig, gridConfig);
    this.axisConfig.validateAxis(type);
    this.axisScale = new CalendarAxisScale(this.axisConfig);
    this.axis = undefined;
  }

  componentDidMount() {
    const pos = this.axisConfig.get('position');
    const { vislibData, renderComplete } = this.props;
    if (pos === 'top') {
      this._drawTop(vislibData);
    } else if (pos === 'left') {
      this._drawLeft(vislibData);
    }
    renderComplete();
  }

  render() {
    const type = this.axisConfig.get('scale.type');
    let axisId;
    if (type === AXIS_SCALE_TYPE.MONTHS) {
      axisId = 'month-labels';
    } else if(type === AXIS_SCALE_TYPE.WEEKS) {
      axisId = 'weekday-labels';
    } else if (type === AXIS_SCALE_TYPE.HOURS) {
      axisId = 'hours-labels';
    } else if (type === AXIS_SCALE_TYPE.MERIDIEM) {
      axisId = 'meridiem-labels';
    }
    return (
      <svg id={axisId} ref={e => {this.axis = e;}} />
    );
  }

  componentWillUnmount() {
    d3.select(this.axis).remove();
    this.axisConfig = null;
    this.axisScale = null;
    this.axis = null;
  }

  _drawTop(vislibData) {
    const [
      type,
      cellSize,
      xOffset,
      yOffset,
      padding
    ] = this.axisConfig.get(['scale.type', 'cellSize', 'xOffset', 'yOffset', 'padding']);

    const axis = d3.select(this.axis);

    if (type === AXIS_SCALE_TYPE.MONTHS) {
      const label = vislibData.label;
      const year = label.slice(0, 4);

      const monthLeftPad = [];
      const [startMonth, endMonth] = this.axisScale.getExtents(vislibData);
      const sMonth = this.axisScale.getNumericScale(startMonth);
      const eMonth = this.axisScale.getNumericScale(endMonth);
      this.axisScale.setExtents({ scaleMin: startMonth, scaleMax: endMonth });
      const startWeek = moment(new Date(year, sMonth - 1, 1)).week() % 52;

      for(let i = sMonth; i <= eMonth; ++i) {
        const pad = cellSize * (
          (i - sMonth) * 1.5 +
          (moment(new Date(year, i - 1, 1)).week() - startWeek) % 52
        );
        monthLeftPad.push(pad);
      }

      this.axisScale.extents.forEach((d, i) => {
        axis.append('text')
          .attr('class', 'month-label')
          .attr('data-month', `${sMonth + i}`)
          .attr('data-year', year)
          .attr('x', monthLeftPad[i] + padding + xOffset * 2)
          .attr('y', yOffset * 2.5)
          .style('font-size', cellSize * 7 / 10)
          .text(d);
      });

      this.props.dispatcher.newMapping().addSource('.month-label').addDataTarget('[data-month]')
        .addEvent('click', expandView);
    } else if (type === AXIS_SCALE_TYPE.WEEKS) {
      this.axisScale.extents.forEach(function (d, i) {
        axis.append('text')
          .attr('class', 'weekday-label')
          .attr('x', ((i + 0.15) * padding) + xOffset * 2)
          .attr('y', yOffset * 2.5)
          .style('font-size', cellSize * 4 / 10)
          .text(d);
      });
    } else if (type === AXIS_SCALE_TYPE.HOURS) {
      this.axisScale.extents.forEach(function (d, i) {
        axis.append('text')
          .attr('class', 'hour-label')
          .attr('x', i * padding + xOffset * 2)
          .attr('y', yOffset * 2.5)
          .style('font-size', cellSize * 3 / 10)
          .text(d);
      });
    }
  }

  _drawLeft() {
    const [type, cellSize, padding, yOffset] = this.axisConfig.get(['scale.type', 'cellSize', 'padding', 'yOffset']);
    const axis = d3.select(this.axis);

    if(type === AXIS_SCALE_TYPE.WEEKS) {
      this.axisScale.extents.forEach(function (d, i) {
        axis.append('text')
          .attr('class', 'weekday-label')
          .attr('y', () => (yOffset * 3.1) + ((i + 0.4) * padding))
          .attr('x', 5)
          .style('font-size', cellSize * 7 / 10)
          .text(d);
      });
    } else if (type === AXIS_SCALE_TYPE.MERIDIEM) {
      this.axisScale.extents.forEach(function (d, i) {
        axis.append('text')
          .attr('class', 'meridiem-label')
          .attr('y', () => (yOffset * 3.2) + ((i + 0.4) * padding))
          .attr('x', 5)
          .style('font-size', cellSize * 3 / 10)
          .text(d);
      });
    }
  }

}
