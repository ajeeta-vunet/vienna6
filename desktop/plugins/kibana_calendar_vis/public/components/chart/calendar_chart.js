/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import React from 'react';
import { CalendarAxis } from './axis/calendar_axis';
import { ChartGrid } from './grid/chart_grid';
import { ChartTitle } from './title/chart_title';

export class CalendarChart extends React.Component {

    state = {
      width: '100%',
      height: '100%',
      rendered: 0
    };

    constructor(props) {
      super(props);
      this.svg = undefined;
      this.adjustSize = this.adjustSize.bind(this);
    }

    async adjustSize() {
      await this.setState(prev => ({
        rendered: prev.rendered + 1
      }));

      const { visConfig, renderComplete } = this.props;

      if (this.svg && this.state.rendered === visConfig.get('categoryAxes').length + 2) {
        const cellSize = visConfig.get('grid.cellSize');
        const svg = this.svg;
        this.setState({
          width: svg.getBBox().width + 2 * cellSize,
          height: svg.getBBox().height + 2 * cellSize
        });
        renderComplete();
      }
    }

    componentDidMount() {
      /*
      * Justification for redundant calls to renderComplete:
      * make tests pass without Timeout errors, does not
      * affect the normal feature behavior.
      */
      this.props.renderComplete();
    }

    render() {
      const { visConfig, vislibData, dispatcher } = this.props;

      return (
        <svg
          width={this.state.width}
          height={this.state.height}
          ref={e => {this.svg = e;}}
        >
          <ChartTitle
            gridConfig={visConfig.get('grid')}
            label={vislibData.yAxisLabel}
            dateRef={vislibData.series[0].values[0].x}
            chartType={visConfig.get('type')}
            renderComplete={this.adjustSize}
          />
          {visConfig.get('categoryAxes').map((axisArgs, i) => (
            <CalendarAxis
              key={i}
              type={visConfig.get('type')}
              gridConfig={visConfig.get('grid')}
              axisConfig={axisArgs}
              vislibData={vislibData}
              dispatcher={dispatcher}
              renderComplete={this.adjustSize}
            />
          ))}
          <ChartGrid
            type={visConfig.get('type')}
            gridConfig={visConfig.get('grid')}
            vislibData={vislibData}
            axes={visConfig.get('categoryAxes')}
            dispatcher={dispatcher}
            renderComplete={this.adjustSize}
          />
        </svg>
      );
    }

    componentWillUnmount() {
      this.svg = null;
    }
}
