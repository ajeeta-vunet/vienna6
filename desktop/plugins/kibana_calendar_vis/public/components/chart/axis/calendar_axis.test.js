/* eslint-disable */

/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import React from 'react';
import moment from 'moment';
import { findDOMNode } from 'react-dom';
import { mount } from 'enzyme';
import sinon from 'sinon';
import { CalendarAxis } from './calendar_axis';
import { CalendarVisConfig } from '../../../lib/calendar_vis_config';
import { AXIS_SCALE_TYPE } from './axis_scale';
import aggResponse from '../../../__tests__/agg_response.json';
import { defaultParams, dayViewParams } from '../../../default_settings';
import { Dispatcher } from '../../../lib';
import { CalendarChart } from '../calendar_chart';
import { calculateBounds } from 'ui/timefilter/get_time';

sinon.spy(CalendarAxis.prototype, 'componentDidMount');

describe('CalendarAxis - default', () => {

  const key = 0;
  let visConfig;
  let axesConfig;
  let visData;
  let dispatcher;
  const fakeVis = {
    params: defaultParams,
    API: {
      timeFilter: {
        getBounds: function () {
          return calculateBounds({
            from: moment(1531026000000),
            to: moment(1531026000000 + 35 * 86400000)
          });
        }
      }
    }
  };

  beforeEach(() => {
    visData = aggResponse.rows[0];
    dispatcher = new Dispatcher().addContainer(document.createElement('div'));
  });

  afterEach(() => {
    visConfig = null;
    visData = null;
    axesConfig = null;
  });

  it('should render a month axis', () => {
    visConfig = new CalendarVisConfig(fakeVis, defaultParams);
    axesConfig = visConfig.get('categoryAxes');
    const axisArgs = axesConfig.filter(axis => axis.scale.type === AXIS_SCALE_TYPE.MONTHS)[0];
    const adjustSize = sinon.spy(CalendarChart.prototype.adjustSize);

    const axisWrapper = mount(<CalendarAxis
      key={key}
      type={visConfig.get('type')}
      gridConfig={visConfig.get('grid')}
      axisConfig={axisArgs}
      vislibData={visData}
      dispatcher={dispatcher}
      renderComplete={adjustSize}
    />);

    expect(CalendarAxis.prototype.componentDidMount.called).toEqual(true);
    expect(adjustSize.calledOnce).toEqual(true);
    const axis = axisWrapper.instance();
    expect(findDOMNode(axis)).toMatchSnapshot();
  });

  it('should render a week axis', () => {
    visConfig = new CalendarVisConfig(fakeVis, defaultParams);
    axesConfig = visConfig.get('categoryAxes');
    const axisArgs = axesConfig.filter(axis => axis.scale.type === AXIS_SCALE_TYPE.WEEKS)[0];
    const adjustSize = sinon.spy(CalendarChart.prototype.adjustSize);

    const axisWrapper = mount(
      <CalendarAxis
        key={key}
        type={visConfig.get('type')}
        gridConfig={visConfig.get('grid')}
        axisConfig={axisArgs}
        vislibData={visData}
        dispatcher={dispatcher}
        renderComplete={adjustSize}
      />
    );

    expect(CalendarAxis.prototype.componentDidMount.called).toEqual(true);
    expect(adjustSize.calledOnce).toEqual(true);
    const axis = axisWrapper.instance();
    expect(findDOMNode(axis)).toMatchSnapshot();
  });

  it('should render an hour axis', () => {
    visConfig = new CalendarVisConfig(fakeVis, dayViewParams);
    axesConfig = visConfig.get('categoryAxes');
    const axisArgs = axesConfig.filter(axis => axis.scale.type === AXIS_SCALE_TYPE.HOURS)[0];
    const adjustSize = sinon.spy(CalendarChart.prototype.adjustSize);

    const axisWrapper = mount(
      <CalendarAxis
        key={key}
        type={visConfig.get('type')}
        gridConfig={visConfig.get('grid')}
        axisConfig={axisArgs}
        vislibData={visData}
        dispatcher={dispatcher}
        renderComplete={adjustSize}
      />
    );

    expect(CalendarAxis.prototype.componentDidMount.called).toEqual(true);
    expect(adjustSize.calledOnce).toEqual(true);
    const axis = axisWrapper.instance();
    expect(findDOMNode(axis)).toMatchSnapshot();
  });

  it('should render a meridiem axis', () => {
    visConfig = new CalendarVisConfig(fakeVis, dayViewParams);
    axesConfig = visConfig.get('categoryAxes');
    const axisArgs = axesConfig.filter(axis => axis.scale.type === AXIS_SCALE_TYPE.MERIDIEM)[0];
    const adjustSize = sinon.spy(CalendarChart.prototype.adjustSize);

    const axisWrapper = mount(
      <CalendarAxis
        key={key}
        type={visConfig.get('type')}
        gridConfig={visConfig.get('grid')}
        axisConfig={axisArgs}
        vislibData={visData}
        dispatcher={dispatcher}
        renderComplete={adjustSize}
      />
    );

    expect(CalendarAxis.prototype.componentDidMount.called).toEqual(true);
    expect(adjustSize.calledOnce).toEqual(true);
    const axis = axisWrapper.instance();
    expect(findDOMNode(axis)).toMatchSnapshot();
  });

});
