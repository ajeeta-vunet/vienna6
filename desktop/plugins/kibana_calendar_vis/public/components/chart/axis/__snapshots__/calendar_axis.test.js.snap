// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarAxis - default should render a meridiem axis 1`] = `
<svg
  id="meridiem-labels"
>
  <text
    class="meridiem-label"
    style="font-size: 12;"
    x="5"
    y="80.8"
  >
    AM
  </text>
  <text
    class="meridiem-label"
    style="font-size: 12;"
    x="5"
    y="122.8"
  >
    PM
  </text>
</svg>
`;

exports[`CalendarAxis - default should render a month axis 1`] = `
<svg
  id="month-labels"
>
  <text
    class="month-label"
    data-month="7"
    data-year="2018"
    style="font-size: 10.5;"
    x="85"
    y="50"
  >
    Jul
  </text>
</svg>
`;

exports[`CalendarAxis - default should render a week axis 1`] = `
<svg
  id="weekday-labels"
>
  <text
    class="weekday-label"
    style="font-size: 10.5;"
    x="5"
    y="68.3"
  >
    Sun
  </text>
  <text
    class="weekday-label"
    style="font-size: 10.5;"
    x="5"
    y="84.05"
  >
    Mon
  </text>
  <text
    class="weekday-label"
    style="font-size: 10.5;"
    x="5"
    y="99.8"
  >
    Tue
  </text>
  <text
    class="weekday-label"
    style="font-size: 10.5;"
    x="5"
    y="115.55"
  >
    Wed
  </text>
  <text
    class="weekday-label"
    style="font-size: 10.5;"
    x="5"
    y="131.3"
  >
    Thu
  </text>
  <text
    class="weekday-label"
    style="font-size: 10.5;"
    x="5"
    y="147.05"
  >
    Fri
  </text>
  <text
    class="weekday-label"
    style="font-size: 10.5;"
    x="5"
    y="162.8"
  >
    Sat
  </text>
</svg>
`;

exports[`CalendarAxis - default should render an hour axis 1`] = `
<svg
  id="hours-labels"
>
  <text
    class="hour-label"
    style="font-size: 12;"
    x="40"
    y="50"
  >
    0:00
  </text>
  <text
    class="hour-label"
    style="font-size: 12;"
    x="80"
    y="50"
  >
    1:00
  </text>
  <text
    class="hour-label"
    style="font-size: 12;"
    x="120"
    y="50"
  >
    2:00
  </text>
  <text
    class="hour-label"
    style="font-size: 12;"
    x="160"
    y="50"
  >
    3:00
  </text>
  <text
    class="hour-label"
    style="font-size: 12;"
    x="200"
    y="50"
  >
    4:00
  </text>
  <text
    class="hour-label"
    style="font-size: 12;"
    x="240"
    y="50"
  >
    5:00
  </text>
  <text
    class="hour-label"
    style="font-size: 12;"
    x="280"
    y="50"
  >
    6:00
  </text>
  <text
    class="hour-label"
    style="font-size: 12;"
    x="320"
    y="50"
  >
    7:00
  </text>
  <text
    class="hour-label"
    style="font-size: 12;"
    x="360"
    y="50"
  >
    8:00
  </text>
  <text
    class="hour-label"
    style="font-size: 12;"
    x="400"
    y="50"
  >
    9:00
  </text>
  <text
    class="hour-label"
    style="font-size: 12;"
    x="440"
    y="50"
  >
    10:00
  </text>
  <text
    class="hour-label"
    style="font-size: 12;"
    x="480"
    y="50"
  >
    11:00
  </text>
  <text
    class="hour-label"
    style="font-size: 12;"
    x="520"
    y="50"
  >
    12:00
  </text>
</svg>
`;
